artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/ActiveMQServerControlImpl.java

```diff
@@ -4721,6 +4722,22 @@ public long getAuthorizationFailureCount() {
       return server.getSecurityStore().getAuthorizationFailureCount();
    }
 
+   @Override
+   public void exportConfigAsProperties() throws Exception {
+      if (AuditLogger.isBaseLoggingEnabled()) {
+         AuditLogger.exportConfigAsProperties(this.server);
+      }
+
+      try (AutoCloseable lock = server.managementLock()) {
+         if (System.getProperty(TMP_DIR_SYSTEM_PROPERTY) == null) {
+            throw new IllegalStateException(TMP_DIR_SYSTEM_PROPERTY + " system property must be set");
+         } else {
+            File exportFileInTmp = new File(System.getProperty(TMP_DIR_SYSTEM_PROPERTY), CONFIG_AS_PROPERTIES_FILE);
```

Comment:
Why not simply print the export and then users can do whatever they want with it (e.g. pipe it to a file, parse it in memory, etc.)?

Reply:
Mostly for security. I don't want to leak any information, config can be sensitive.
In addition it could be quite large.
Doing transformation in place is safest

Reply:
We will be obfuscating sensitive (e.g. password) fields, no? Are there any security implications with putting the file into `/tmp`?

Reply:
Yes, but I don't think that is sufficient.
instance-dir/tmp is the default,  instance-dir/etc would be best but could or should be read-only